File: questions/Q3355.go
package questions

func isZeroArray(nums []int, queries [][]int) bool {
	if len(nums) == 0 {
		return false
	}

	differ := make([]int, len(nums)+1)
	for _, query := range queries {
		differ[query[0]]++
		differ[query[1]+1]--
	}

	sum := 0
	for i := 0; i < len(nums); i++ {
		sum += differ[i]
		if nums[i] > sum {
			return false
		}
	}
	return true
}

File: questions/Q3355.java
package questions;

public class Q3355 {
    
    public boolean isZeroArray(int[] nums, int[][] queries) {
        if (nums == null || nums.length == 0 || queries == null || queries.length == 0) {
            return false;
        }

        int[] differ = new int[nums.length+1];
        for (int i = 0; i < queries.length; i++) {
            int left = queries[i][0];
            int right = queries[i][1];
            differ[left]++;
            differ[right + 1]--;
        }
        for(int i = 0,sum = 0;i<nums.length;i++){
            sum += differ[i];
            if (nums[i] > sum) {
                return false;
            }
        }
        return true;
    }

}
